### PR TITLE
Allow setting anonymous tracks user ID when redirecting into authorize flow

### PR DESCRIPTION
**Pull request by gravityrail**
Builds on changes in w.js in D7080-code which allow setting the anonId to a new value without losing aliasing of prior events.

To test: 
- Apply D7080-code to your WPCOM sandbox
- Go through Jetpack Connect auth flow and log in
- You should observe calls to pixel.wp.com/t.gif which alias anon user `jetpack:XXXXXX` to the WPCOM user, AND the existing, Calypso-generated anonId
- The `ai` cookie should be set to the `jetpack:XXXXXX` value

### CODE REVIEW

File: client/jetpack-connect/authorize-form.jsx [hunk 1]
```diff
@@ -67,6 +68,12 @@ class JetpackConnectAuthorizeForm extends Component {
 	};
 
 	componentWillMount() {
+		// set anonymous ID for cross-system analytics
+		const userId = this.props.jetpackConnectAuthorize.queryObject._ui;
```
[the remark this answers is not in the document]
Comment (gravityrail) [reply client/jetpack-connect/authorize-form.jsx]: Fixed

File: client/state/analytics/actions.js [hunk 2]
```diff
@@ -63,6 +64,16 @@ export const recordEvent = ( service, args ) => ( {
 	}
 } );
 
+export const setTracksAnonymousUserId = ( anonId ) => ( {
```
[the remark this answers is not in the document]
Comment (gravityrail) [reply client/state/analytics/actions.js]: Fixed

File: client/state/analytics/middleware.js [hunk 3]
```diff
@@ -56,6 +57,9 @@ export const dispatcher = ( { meta: { analytics: analyticsMeta } }, state ) => {
 			case ANALYTICS_PAGE_VIEW_RECORD:
 				return invoke( pageViewServices, service, payload );
 
+			case ANALYTICS_TRACKS_ANONID_SET:
+				return analytics.tracks.setAnonymousUserId( payload );
```
[the remark this answers is not in the document]
Comment (gravityrail) [reply client/state/analytics/middleware.js]: Fixed

**Comment by gravityrail** [comment]
@tyxla - I tried to address your feedback (thank you!). Could you take another look?

**Comment by ebinnion** [comment]
In testing, I could confirm that the `_ui` query object param was being passed. I wasn't sure where to look to verify that the value was being sent to Tracks though. I did not see an `ai` cookie with `jetpack:*` nor did I see any `pixel.wp.com/t.gif` requests with `jetpack:*` as an argument. It could be that I was looking at the wrong thing though.

That being said, this did not break the connection flow for me. So, if you're confident that this is tracking as it should, 🚢 .

**Pull request merged**